app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt

```diff
@@ -115,6 +158,583 @@ class AdministratorControlsFragmentTest {
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "sw600dp")
+  fun testAdministratorControls_defaultTabletConfig_profileListIsDisplayed() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 0, R.id.profile_list_name)).check(
+        matches(withText("Admin"))
+      )
+      onView(
+        atPositionOnView(
+          R.id.profile_list_recycler_view,
+          0,
+          R.id.profile_list_admin_text
+        )
+      ).check(
+        matches(withText(context.resources.getString(R.string.profile_chooser_admin)))
+      )
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name)).check(
+        matches(withText("Ben"))
+      )
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "sw600dp")
+  fun testAdministratorControls_tabletConfigChange_profileListIsDisplayed() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 0, R.id.profile_list_name)).check(
+        matches(withText("Admin"))
+      )
+      onView(
+        atPositionOnView(
+          R.id.profile_list_recycler_view,
+          0,
+          R.id.profile_list_admin_text
+        )
+      ).check(
+        matches(withText(context.resources.getString(R.string.profile_chooser_admin)))
+      )
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name)).check(
+        matches(withText("Ben"))
+      )
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "sw600dp")
+  fun testAdministratorControls_selectProfileAdmin_backButton_selectSecondProfileDisplayed() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 0, R.id.profile_list_name)).check(
+        matches(withText("Admin"))
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.extra_controls_title)).check(matches(withText("Admin")))
+      onView(withId(R.id.profile_edit_name)).check(matches(withText("Admin")))
+      onView(withId(R.id.administrator_controls_multipane_options_back_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name))
+        .check(matches(withText("Ben")))
+      onView(atPositionOnView(R.id.profile_list_recycler_view, 1, R.id.profile_list_name))
```

Comment:
i have broken down the implementation

Reply:
Now this test is verifying multiple behaviors--you should split it up such that each test has just one arrange/act/assert block (in that order). Multiple act/assert blocks in a single test indicates more than one behavior is being tested, and we should aim to verify only a single behavior for each test method.

Reply:
actuallly these tests are big as I am navigation with the profiles and buttons I am not directly able to inflate ProfileEditFragment and then see things they are very closely coupled.

Reply:
I'm not sure that I'm understanding what the issue is. I left a bit more detailed comment on this suggestion above, but the main thing we should avoid is performing multiple assertions that are unrelated since that indicates multiple behaviors being tested.

The tests themselves can be shortened by leveraging helper methods to consolidate common setup to simple-to-use methods.

Reply:
made some helper functions to shorten the tests.